tests/unit_tests/optional.py

```diff
@@ -0,0 +1,67 @@
+import pytest
```

Comment:
You should name this file `test_optional.py` so that the testing tools pick it up automatically. 😄

Reply:
Good tests for the optional enhancement!